addons/account/models/account_move.py

```diff
@@ -1864,7 +1864,13 @@ def copy(self, default=None):
             default['date'] = self.company_id._get_user_fiscal_lock_date() + timedelta(days=1)
         if self.move_type == 'entry':
             default['partner_id'] = False
-        return super(AccountMove, self).copy(default)
+        copied_am = super().copy(default)
+        if self.state != 'draft':
+            copied_am._message_log(body=_(
+                'This entry has been duplicated from %s',
+                html_escape(self.name)
```

Comment:
Better use `self.display_name` and remove the condition on `self.state`

Reply:
@tsb-odoo says he prefers no log if the original entry is still in draft

Reply:
Mmmh, okay then. I will let him test before merging. :+1:

Reply:
Thanks, both suggestions were incorporated:

1. The message includes a clickable link
2. The `display_name` is improved with a bare minimum of (* [id])